### ISSUE TITLE
`parse_gcov` assumes that produces source file names are escaped regular expressions

### ISSUE THREAD

**Issue by hsloot** [r-lib/covr#459]
The function `parse_gcov` tries to skip all files for which `source_file` is not below root `package_path`. It does that by using a regular expression. 

https://github.com/r-lib/covr/blob/ebf0e7fc215ef91aeeb43ffcbe945e116e6e0187/R/compiled.R#L14

However, `rex::regex` assumes that the given expression is already an escaped regular expression. That might not be always the case - MacOS is allowing almost all Unicode characters in filenames. In my case, a parent folder contains a "+" (that, unfortunately, cannot be changed). 

It should be able to solve this by replacing the relevant expression with 

```r
grepl(rex::rex(start, rex::regex(rex::escape(paste0(package_path, collapse = "|")))), source_file)
```

Possible link to #448.

**Comment by jimhester** [comment]
Yes, your fix looks correct, would you be interested in submitting a pull request to fix this?
